src/lib/Components/Bidding/Screens/SelectCountry.tsx

```diff
@@ -0,0 +1,141 @@
+import { Container } from "../Components/Containers"
+import { Input } from "../Components/Input"
+import { Title } from "../Components/Title"
+import { Serif18, SerifItalic18 } from "../Elements/Typography"
+import { Country, SearchResult } from "../types"
+
+const { Emission } = NativeModules
+
+interface SelectCountryProps extends ViewProperties {
+  navigator: NavigatorIOS
+  onCountrySelected?: (country: any) => void
+}
+
+interface SelectCountryState {
+  query: string
+  isLoading: boolean
+  results: SearchResult[]
+}
+
+// TODO: Remove this once React Native has been updated
+const isPhoneX = Dimensions.get("window").height === 812 && Dimensions.get("window").width === 375
+const defaultVerticalOffset = isPhoneX ? 30 : 15
+
+export class SelectCountry extends React.Component<SelectCountryProps, SelectCountryState> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      query: null,
+      isLoading: false,
+      results: [],
+    }
+  }
+
+  locationSelected = async (result: SearchResult) => {
+    // The extra key is from a throwaway google account,
+    // it should only get used in Emission dev mode
+    const apiKey = Emission.googleMapsAPIKey || "AIzaSyBJRIy_zCXQ7XYt9Ubn8bpUIEAxEOKUmx8"
+    const queryString = stringify({
+      key: apiKey,
+      placeid: result.id,
+    })
+
+    const response = await fetch(`https://maps.googleapis.com/maps/api/place/details/json?${queryString}`)
+    const results = await response.json()
+    const country = results.result.address_components.find(comp => comp.types[0] === "country")
+
+    this.props.onCountrySelected({ longName: country.long_name, shortName: country.short_name } as Country)
+    this.props.navigator.pop()
+  }
+
+  textChanged = async (text: string) => {
+    this.setState({ query: text, isLoading: text.length > 0 })
+    this.searchForQuery(text)
+  }
+
+  // https://developers.google.com/places/
+  // https://developers.google.com/places/web-service/details
+  searchForQuery = async (input: string) => {
+    // The 2nd is a throw-away key on a throw-away account
+    const apiKey = Emission.googleMapsAPIKey || "AIzaSyBJRIy_zCXQ7XYt9Ubn8bpUIEAxEOKUmx8"
+    const queryString = stringify({
+      key: apiKey,
+      language: "en",
+      types: "(regions)",
+      input,
+    })
+
+    const response = await fetch(`https://maps.googleapis.com/maps/api/place/autocomplete/json?${queryString}`)
+    const results = await response.json()
+    const predictions = (results.predictions || [])
+      .filter(prediction => prediction.types[0] === "country")
+      .map(prediction => ({ id: prediction.place_id, name: prediction.description }))
+
+    this.setState({
+      results: predictions,
+      isLoading: false,
+    })
+  }
+
+  render() {
+    const { results, isLoading, query } = this.state
+
+    return (
+      <BiddingThemeProvider>
+        <KeyboardAvoidingView behavior="padding" keyboardVerticalOffset={defaultVerticalOffset} style={{ flex: 1 }}>
```

Comment:
From the gif you posted it looks like everything is sufficiently above the keyboard and oriented towards the top of the screen. Is adding this wrapper necessary for being able to display all of the autocomplete results or can we get rid of it?

Reply:
Good call, let me test this without it.